src/main/java/net/glowstone/entity/GlowPlayer.java

```diff
@@ -707,6 +710,7 @@ public void remove() {
             scoreboard.unsubscribe(this);
             scoreboard = null;
         }
+        clearBossBars();
```

Comment:
Why are boss bars being updated for a player that was removed from the server?

Reply:
To avoid memory leaks resulting from the player still holding a reference to the boss bar, or vice-versa.

Reply:
I guess this is an acceptable solution for now, but later on we may want to investigate using weaker references to prevent the problem.